### DISCUSSION TITLE
用OpenCV將影片的每個frame轉成圖片

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS4719-C/Python-Community-GS4719-C#428] [General]
因為最近在標記(label)訓練機器學習模型用的資料集，當中需要把影片的每個frame存成圖片，所以找到了透過Python來達成這件事的方法:

1. 安裝OpenCV (以前有安裝過的話這步就不用了)
`python3 -m pip install opencv-python`
2. 執行以下程式
(影片檔需命名為`video.mp4`並與程式碼放在同一個資料夾下)
```python
import cv2
vidcap = cv2.VideoCapture('video.mp4') # source video
success, image = vidcap.read()
while success:
    cv2.imwrite("frame_"+"{:06n}".format(count)+".PNG",
                image)     # save frame as PNG file
    success, image = vidcap.read()
```
3. 圖片檔就產生囉~
<img width="1431" alt="Screen Shot 2022-03-09 at 14 00 00" src="https://user-images.githubusercontent.com/32983347/157381636-55041796-e3a3-4ee3-80b9-dc5b8128c414.png">
 
原本約11MB的影片檔，竟然產生了約600MB的圖片檔，可見mp4的壓縮率真的很強！大家也可以試試看把code改成每幾個frame才存成圖片~

**Comment by ghost** [comment]
這邊稍微補充一下單純撥放影片的code：
```
# 讀取影片 並依據waitkey1撥放(但無聲音
cap = cv2.VideoCapture('VIVACE.mp4')  # 如果寫1or0則為外接攝像頭與前置鏡頭

while True:
    ret, frame = cap.read()  # 下一幀圖片有沒有回傳成功，此回傳圖片
    if ret:
        frame = cv2.resize(frame, (0, 0), fx=0.5, fy=0.5)
        cv2.imshow('video', frame)
    else:
        break

    if cv2.waitKey(26) == ord('q'):
        break
```
關於幀這個概念也建議想了解這塊的同學稍微查一下相關資料

**Reply by gnsJhenJie** [reply]
在打markdown 程式碼區塊的時候可以把第一個「 \`\`\`」 改成「\`\`\`python」就能幫python程式碼上色了~

**Reply by chia-kai-chang** [reply]
像這樣~
![image](https://user-images.githubusercontent.com/1768891/157444140-5af1a6dc-b33d-4069-8e59-5cdc8425543a.png)

**Comment by chupochun** [comment]
原來還有這樣的方式可以將影片轉成圖片

**Comment by sean1325418** [comment]
若是想要從影片中擷取圖像做圖像分析，那這程式還滿方便的

**Comment by xkllkx** [comment]
謝謝分享，這對圖像辨識的素材取得十分有幫助

**Comment by ThomasMeng0515** [comment]
opencv對於標記訓練機器模型的圖片真的很有用~
